docs/internals_gpio_control.md

```diff
@@ -6,17 +6,17 @@ QMK has a GPIO control abstraction layer which is microcontroller agnostic. This
-| `writePinHigh(pin)`    | Set pin level as high, assuming it is an output  | `PORTB \|= (1<<2)`                              | `palSetLine(pin)`                               |
-| `writePinLow(pin)`     | Set pin level as low, assuming it is an output   | `PORTB &= ~(1<<2)`                              | `palClearLine(pin)`                             |
-| `writePin(pin, level)` | Set pin level, assuming it is an output          | `(level) ? PORTB \|= (1<<2) : PORTB &= ~(1<<2)` | `(level) ? palSetLine(pin) : palClearLine(pin)` |
-| `readPin(pin)`         | Returns the level of the pin                     | `_SFR_IO8(pin >> 4) & _BV(pin & 0xF)`           | `palReadLine(pin)`                              |
-| `togglePin(pin)`       | Invert pin level, assuming it is an output       | `PORTB ^= (1<<2)`                               | `palToggleLine(pin)`                            |
+|Function                |Description                                       | Old AVR Examples                                    | Old ChibiOS/ARM Examples                        |
+|------------------------|--------------------------------------------------|-----------------------------------------------------|-------------------------------------------------|
+| `setPinInput(pin)`     | Set pin as input with high impedance (High-Z)    | `DDRB &= ~(1<<pin)`                                 | `palSetLineMode(pin, PAL_MODE_INPUT)`           |
+| `setPinInputHigh(pin)` | Set pin as input with builtin pull-up resistor   | `DDRB &= ~(1<<pin); PORTB \|= (1<<pin)`             | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)`    |
+| `setPinInputLow(pin)`  | Set pin as input with builtin pull-down resistor | N/A (Not supported on AVR)                          | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)`  |
+| `setPinOutput(pin)`    | Set pin as output                                | `DDRB \|= (1<<pin)`                                 | `palSetLineMode(pin, PAL_MODE_OUTPUT_PUSHPULL)` |
+| `writePinHigh(pin)`    | Set pin level as high, assuming it is an output  | `PORTB \|= (1<<pin)`                                | `palSetLine(pin)`                               |
+| `writePinLow(pin)`     | Set pin level as low, assuming it is an output   | `PORTB &= ~(1<<pin)`                                | `palClearLine(pin)`                             |
+| `writePin(pin, lvl)`   | Set pin level, assuming it is an output          | `(lvl) ? PORTB \|= (1<<pin) : PORTB &= ~(1<<pin)`   | `(lvl) ? palSetLine(pin) : palClearLine(pin)`   |
+| `readPin(pin)`         | Returns the level of the pin                     | `_SFR_IO8(pin >> 4) & _BV(pin & 0xF)`               | `palReadLine(pin)`                              |
+| `togglePin(pin)`       | Invert pin level, assuming it is an output       | `PORTB ^= (1<<pin)`                                 | `palToggleLine(pin)`                            |
```

Comment:
Could you revert (well, rather, pull out) these changes? 

If you could, open a separate PR with just this.